### ISSUE TITLE
IP Banning Users?

### ISSUE THREAD

**Issue by TylerCode** [discuitnet/discuit#10]
IP banning users is not something that is going to happen often, but might be worth implementing to empower the site admins to perform them. There is currently a user who is a likely candidate but as the site grows there will be others and having that power sit with only the host (banning using iptables etc) will likely cause delays in actions being taken. 

If the team didn't feel comfortable giving any one admin that power, maybe it's worth making it an automatic process based on the number of accounts banned on the same IP? 

Ban evasions hurt the community in many ways, between the continued actions of whatever the user got banned for in the first place to the arbitrary exhaustion of usernames. 

If this is already a solved problem, then feel free to ban me from contributing 😄, I did poke through the code though and saw that IP tracking happened but didn't see anywhere that the IP address was used to ban a user. 

Thanks for reading!

**Comment by previnder** [comment]
Yeah, this is a good idea; thanks for the suggestion. So far, the few times we had to IP ban someone, it was done at the firewall. And that means only someone with SSH access to the server could do it (right now, that's just me). Currently, all admin accounts have access to everything else besides this: ban users, remove anything, change mods, etc.

**Comment by TylerCode** [comment]
Yeah, I figured the existing IP bans were done with the firewall or IPtables or something. Just figured it was worthwhile after talking to one of the admins, though, the specific user I had in mind seems to use VPNs anyway so.

**Comment by Oracuda** [comment]
It should be optional for community admins and further overrideable by site admins. There's some edge cases involved here for people living in the same WIFI and probably more realistic; CGNAT.
